restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/client/sdk/RCConnection.java

```diff
@@ -560,10 +563,19 @@ void initializeWebrtc(boolean videoEnabled)
         for (String permission : MANDATORY_PERMISSIONS) {
             if (context.checkCallingOrSelfPermission(permission) != PackageManager.PERMISSION_GRANTED) {
                 logAndToast("Permission " + permission + " is not granted");
-                // TODO: return error to RCConnection listener
+                listener.onPermissionNotGranted(this, permission);
```

Comment:
@mehturt: had a better look at the code. Ok so actually we don't want to return null from connect() in that case. Null is for cases where the RCConnection objects wasn't even initialized. But in this case the RCConnection is properly initialized and after that we realize we have no permissions. Hence, the application can use the RCConnection object if it wants without issues; it just won't be 'connected' until we get the error. It doesn't matter if this happens synchronously, we still should use the error reporting via the callback (i.e. onDisconnect()) as it gives us a means to convey error code and error text. Please let me know if still unclear.